Android/ThirdPartyAuth/app/src/main/java/com/bentley/sample/thirdpartyauth/LoginViewModel.kt

```diff
@@ -0,0 +1,55 @@
+package com.bentley.sample.thirdpartyauth
+
+import android.content.Context
+import android.content.Intent
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.auth0.android.Auth0
+import com.auth0.android.authentication.AuthenticationException
+import com.auth0.android.callback.Callback
+import com.auth0.android.provider.WebAuthProvider
+import com.auth0.android.result.Credentials
+import com.bentley.sample.shared.MainActivity
+
+class LoginViewModel(
+	resourceHelper: IResourceHelper,
+	onAuth0TokenUpdated: (auth0Token: String)->Unit = {},
+) : ViewModel() {
+	/** Used to lookup string resources */
+	private val _resourceHelper = resourceHelper
+	var onAuth0TokenUpdated = onAuth0TokenUpdated
+
+	/** Authorization client to login to Auth0 */
+	private var _auth0Client: Auth0 = Auth0(
+		resourceHelper.getString(R.string.ITMSAMPLE_AUTH0_CLIENT_ID),
+		resourceHelper.getString(R.string.ITMSAMPLE_AUTH0_DOMAIN)
+	)
+
+	var displayText: MutableLiveData<String> = MutableLiveData("")
+
+	fun loginToAuth0(context: Context) {
+		displayText.value = "Logging into Auth0..."
+		WebAuthProvider.login(_auth0Client)
+			.withScheme(_resourceHelper.getString(R.string.ITMSAMPLE_AUTH0_SCHEME))
+			.withScope("openid profile offline_access")
+			.withAudience(_resourceHelper.getString(R.string.ITMSAMPLE_AUTH0_AUDIENCE))
+			// Launch the authentication passing the callback where the results will be received
+			.start(context, object : Callback<Credentials, AuthenticationException> {
+				// Called when there is an authentication failure
+				override fun onFailure(exception: AuthenticationException) {
+					displayText.value = "Something went wrong receiving token from auth0: ${exception.localizedMessage}"
```

Comment:
This should be localized.

Reply:
Note: when localizing a string that includes a variable, the localized string must be a format into which the variable value is placed (for example using %1$s in the string). The following is not acceptable:
```kotlin
_resourceHelper.getString(R.string.AUTH0_LOGIN_ERROR_PREFIX) + exception.localizedMessage
```